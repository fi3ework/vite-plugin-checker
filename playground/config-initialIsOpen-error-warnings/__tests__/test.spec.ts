import { describe, expect, it } from 'vitest'

import {
  editFile,
  getHmrOverlay,
  getHmrOverlayText,
  isServe,
  pollingUntil,
  sleepForEdit,
  sleepForServerReady,
} from '../../testUtils'

describe('config-initialIsOpen-error-warnings', () => {
  describe.runIf(isServe)('serve', () => {
    it('should not show overlay when only warning exists', async () => {
      await sleepForServerReady()
      try {
        await getHmrOverlayText()
      } catch (e) {
        expect((e as any).toString()).toContain(
          'Invariant failed: <vite-plugin-checker-error-overlay> shadow dom is expected to be found, but got null'
        )
      }

      console.log('-- overlay appears after introduce an error --')
      editFile('src/main.ts', (code) => code.replace('const rootDom', 'var rootDom'))
      await sleepForEdit()
      await getHmrOverlayText()
      await pollingUntil(getHmrOverlay, (dom) => !!dom)
      const [, , frame] = await getHmrOverlayText()
      expect(frame).toMatchSnapshot()
    })
  })
})
