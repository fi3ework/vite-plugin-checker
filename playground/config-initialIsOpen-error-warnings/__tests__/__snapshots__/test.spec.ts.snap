// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`config-initialIsOpen-error-warnings > serve > should not show overlay when only warning exists 1`] = `
"    1 | import { text } from './text'
    2 |
  > 3 | var rootDom = document.querySelector('#root')! as HTMLElement
      | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    4 | rootDom.innerHTML = text
    5 |
    6 | export {}"
`;
